components/brave_wallet_ui/components/desktop/popup-modals/add-account-modal/add-imported-account-modal.tsx

```diff
@@ -0,0 +1,303 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { useHistory, useParams } from 'react-router'
+
+// utils
+import { getLocale } from '$web-common/locale'
+import { copyToClipboard } from '../../../../utils/copy-to-clipboard'
+
+// options
+import { CreateAccountOptions } from '../../../../options/create-account-options'
+
+// types
+import { BraveWallet, CreateAccountOptionsType, PageState, WalletRoutes, WalletState } from '../../../../constants/types'
+import { FilecoinNetworkTypes, FilecoinNetworkLocaleMapping, FilecoinNetwork } from '../../../../common/hardware/types'
+
+// actions
+import { WalletPageActions } from '../../../../page/actions'
+
+// components
+import { Select } from 'brave-ui/components'
+import { DividerLine, NavButton } from '../../../extension'
+import PopupModal from '..'
+import { SelectAccountType } from './select-account-type/select-account-type'
+
+// style
+import {
+  DisclaimerText,
+  ErrorText,
+  ImportButton,
+  ImportDisclaimer,
+  ImportRow,
+  Input,
+  SelectWrapper,
+  StyledWrapper
+} from './style'
+
+interface Params {
+  accountTypeName: string
+}
+
+const reduceFileName = (address: string) => {
+  const firstHalf = address.slice(0, 4)
+  const secondHalf = address.slice(-4)
+  const reduced = firstHalf.concat('......', secondHalf)
+  return reduced
+}
+
+export const ImportAccountModal = () => {
+  // refs
+  const passwordInputRef = React.useRef<HTMLInputElement>(null)
+
+  // routing
+  const history = useHistory()
+  // const { pathname: walletLocation } = useLocation()
+  const { accountTypeName } = useParams<Params>()
+
+  // redux
+  const isSolanaEnabled = useSelector(({ wallet }: { wallet: WalletState }) => wallet.isSolanaEnabled)
+  const isFilecoinEnabled = useSelector(({ wallet }: { wallet: WalletState }) => wallet.isFilecoinEnabled)
```

Comment:
Same here?

Reply:
selectors should only return one value to reduce chances or re-rendering